lib/src/widgets/dashboard_screen/main_drawer.dart

```diff
@@ -0,0 +1,89 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_icons/flutter_icons.dart';
+import 'package:provider/provider.dart';
+import 'package:saraka/blocs.dart';
+import 'package:saraka/constants.dart';
+
+class MainDrawer extends StatefulWidget {
+  State<MainDrawer> createState() => _MainDrawerState();
+}
+
+class _MainDrawerState extends State<MainDrawer> {
+  AuthenticationBloc _authenticationBloc;
+
+  void initState() {
+    super.initState();
+
+    Future.delayed(Duration.zero, () {
+      final authenticationBloc = Provider.of<AuthenticationBloc>(context);
+
+      setState(() {
+        _authenticationBloc = authenticationBloc;
+      });
+    });
+  }
```

Comment:
You don't have to choose to use `StatefulWidget` and `initState` for this case. You can use a `Consumer` widget in `build()` to retrieve a bloc from the ancestor.